exporter/awsemfexporter/metric_translator.go

```diff
@@ -257,9 +259,13 @@ func buildCWMetricFromDP(dp interface{}, pmd *pdata.Metric, namespace string, me
 		fieldsPairs[k] = v.Value()
 		dimensionSlice = append(dimensionSlice, k)
 	})
-	// add OTLib as an additional dimension
-	fieldsPairs[OtlibDimensionKey] = OTLib
-	dimensionArray = append(dimensionArray, append(dimensionSlice, OtlibDimensionKey))
+	// add OTel instrumentation lib name as an additional dimension if it is defined
+	if instrumentationLibName != noInstrumentationLibraryName {
```

Comment:
What happens if we set "unknown", "unset", or similar instead of filtering? Doesn't it allow the metrics to be more symmetric (e.g., when analyzing metrics based on this dimension, all the data will be present instead of having a gap)?

Reply:
The assumption is that the metrics generated by OTel SDKs should not have NonNull `InstrumentationLibrary` and this dimension won't be filtered on these metrics. The complaints are from `AWSECSMetadataReceiver` and `StatsDReceiver` users saying why they are having `OTelLib` dimension in their metrics but they're not using any of instrumentation lib to generate metrics. Thanks